lib/discourse_algolia/indexer.rb

```diff
@@ -33,6 +33,12 @@ def process!(ids: nil)
     @index.delete_objects(ids) if ids.present?
   end
 
+  def queue_ids
+    ids = Discourse.redis.lrange(self::QUEUE_NAME, 0, QUEUE_SIZE - 1)
+    Discourse.redis.ltrim(self::QUEUE_NAME, ids.size, -1)
```

Comment:
Do we need to do this in a "transaction"?

Reply:
You are right, but I am not sure how I can achieve that. I tried to do it in a `multi` block, but it did not work because the result of `lrange` would not be available until the whole block executed.